### DISCUSSION TITLE
Invalid JSON data in request body

### DISCUSSION THREAD

**Discussion by Mingo07-dev** [fattureincloud/api#368] [Q&A] [answered]
I'm trying to create a valid JSON from zero following the guide https://developers.fattureincloud.it/docs/guides/invoice-creation , but I'm getting this error: 
```
"error": {
    "message": "Invalid JSON data in request body.",
    "validation_result": null
  }
```
What I 'm missing, one or more mandatory properties, or am I just writing something wrong?
Moreover I don't understand the difference between due_date and payment_terms inside payment_list, are they the ways to express the same thing?

Here's my JSON:
```
{
    "data": {
        "entity": {
            "name": "Mario Rossi",
            "first_name": "Mario ",
            "last_name": "Rossi",
	    "tax_code": "RSSMAR99R20G273F",
	    "address_street": "Via Libertà",
	    "address_postal_code": "1",
	    "address_city": "Palermo",
	    "address_province": "PA",
             "country": "Italia",
        },
        "type": "invoice",
        "numeration": "FT-2023",
		"currency": {
			"id": "EUR",
			"exchange_rate": "1.00000",
			"symbol": "€"
		},
		"language": {
			"code": "it",
			"name": "Italiano"
		},
        "items_list":{
            "name": "NomeTrattamento",
            "net_price": 30,
            "vat": {
                "id": 45
            },
            "qty": 1
        },
        "payments_list": [
            {
                "amount": 30,
                "due_date": "2023-01-01",
                "status": "not_paid",
                "payment_account": {
                    "id": 1127896
                }
            }
        ],
        "extra_data": {
            "ts_communication": true,
            "ts_pagamento_tracciato": true,
            "ts_tipo_spesa": "SR" 
        }
    }
}
```

**Comment by emajo** [comment, marked as answer]
Hi @Mingo07-dev, the error you are receiving is due to a formal error in the json (in this case `"country": "Italia",` shouldn't contain the trailing comma), also when you are missing a mandatory property in the request body the error that gets returned contains exactly which fields are missing.

About the `due_date` and `payment_terms` question, actually the due_date field is mandatory even if payment_terms is specified (ideally the due_date could be automatically calculated in this case, but it doesn't work like that now), in fact the payment_terms field is there only if you want to keep track of them.

Thanks,
Emanuele.

**Reply by Mingo07-dev** [reply]
Thank you, it works by deleting the comma and I also forgot that items_list must be an array of objects while it was an object

**Reply by valmoz** [reply]
Hi @Mingo07-dev 
We suggest using one of our [SDKs](https://developers.fattureincloud.it/docs/sdks/) to avoid this kind of error when possible.
Have you checked if one of those could be of interest?

Thanks 
Mauro

**Reply by Mingo07-dev** [reply]
Yes, I'm using the PHP SDKs, but I was testing in the API reference to undersand how to compose my PHP array with just the fields that I need
